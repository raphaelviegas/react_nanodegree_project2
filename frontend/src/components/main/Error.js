import React from 'react'
import { Link } from 'react-router-dom'

export default () => {
  return (
    <div className="container center-align error-page">
      <h3 className="error-title">Ops, something went wrong!</h3>
      <p className="error-text">This page doesn't exist</p>
      <Link to="/" className="error-link waves-effect waves-light btn">Go back to the home page</Link>
    </div>
  )
}
