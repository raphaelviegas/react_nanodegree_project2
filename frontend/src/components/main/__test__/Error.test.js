import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import Error from '../Error'
import { Link } from 'react-router-dom'

describe('<Error />', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallow(<Error />)
    })

    it('Should render one .error-page element', () => {
        expect(wrapper.find('.error-page').length).toEqual(1)
    })

    it('Should render one link tag to /', () => {
        expect(wrapper.containsMatchingElement(<Link to="/">Go back to the home page</Link>)).toEqual(true)
    })

    it('Should have one error message title', () => {
        expect(wrapper.find('.error-title').length).toEqual(1)
    })

    it('Should have one error message text', () => {
        expect(wrapper.find('.error-text').length).toEqual(1)
    })

    it('Should render correctly', () => {
        const tree = renderer.create(
            <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
                <Error />
            </MemoryRouter>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

})
