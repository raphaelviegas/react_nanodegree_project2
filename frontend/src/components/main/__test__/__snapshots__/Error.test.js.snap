// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> Should render correctly 1`] = `
<div
  className="container center-align error-page"
>
  <h3
    className="error-title"
  >
    Ops, something went wrong!
  </h3>
  <p
    className="error-text"
  >
    This page doesn't exist
  </p>
  <a
    className="error-link waves-effect waves-light btn"
    href="/"
    onClick={[Function]}
  >
    Go back to the home page
  </a>
</div>
`;
